lib/cartopy/io/shapereader.py

```diff
@@ -435,15 +438,29 @@ def acquire_all_resources(self, format_dict):
 
         # Download archive.
         url = self.url(format_dict)
-        shapefile_online = self._urlopen(url)
+        try:
+            shapefile_online = self._urlopen(url)
+        except HTTPError:
+            warnings.warn(
```

Comment:
Does this mean that if ngdc updates to a new version, users are going to get a warning that they can't fix (without resorting to a warning filter)? I'm not a fan of issuing warnings when users are doing nothing wrong and can't fix it.

Reply:
Yes this is what it means, but ...

- it happens only once (and a Download warning also happens when the coastlines are downloaded anyway)
- it is better than having no warning but the bugs of an old coastline data set.

Alternative would be to either download the older version in the first place as before or leave the warning away.

Reply:
There is a third option, which is what I was trying to hint at, which is to put a version number in somewhere.

```python
_GSHHS_URL_TEMPLATE = f'https://www.ngdc.noaa.gov/mgg/shorelines/data/gshhg/latest/gshhg-shp-{version}.zip'
```
Then on the except you would do:
```python
url = f'https://www.ngdc.noaa.gov/mgg/shorelines/data/gshhs/oldversions/version{version}/gshhg-shp-{version}.zip'
```

I _assume_ that the 2.3.7 will be moved over to the archive at some point when a new one comes up, and this would catch that case assuming they keep their naming convention.

Reply:
Shouldn't we have something in the except that works even if the naming convention is changed?

Reply:
:+1: Add a fallback if both of those fail to the 2.3.6 version if both throw the URL not found error?

Reply:
Okay, here is my attempt